packages/pyright-internal/src/backgroundAnalysisBase.ts

```diff
@@ -339,20 +345,20 @@ export abstract class BackgroundAnalysisRunnerBase extends BackgroundThreadBase
 
             case 'analyzeFile': {
                 run(() => {
-                    const { filePath, cancellationId } = msg.data;
+                    const { fileUri, cancellationId } = msg.data;
                     const token = getCancellationTokenFromId(cancellationId);
 
-                    return this.handleAnalyzeFile(filePath, token);
+                    return this.handleAnalyzeFile(fileUri, token);
                 }, msg.port!);
                 break;
             }
 
             case 'getDiagnosticsForRange': {
                 run(() => {
-                    const { filePath, range, cancellationId } = msg.data;
+                    const { fileUri, range, cancellationId } = msg.data;
                     const token = getCancellationTokenFromId(cancellationId);
 
-                    return this.handleGetDiagnosticsForRange(filePath, range, token);
+                    return this.handleGetDiagnosticsForRange(fileUri, range, token);
```

Comment:
Where does `fileUri` get converted back from a string to a `Uri` object? Seems like it must be happening since `this.handleGetDiagnosticsForRange` takes a `Uri`, but isn't the `fileUri` that was sent in the `msg.data` a string?

Reply:
Oh this is a really good catch. This wouldn't have worked. It wasn't doing the conversion. I guess none of our tests use the background thread.